pkg/rtc/room.go

```diff
@@ -831,10 +831,14 @@ func (r *Room) UpdateParticipantMetadata(
 		}
 	}
 	if metadata != "" {
-		participant.SetMetadata(metadata)
+		if err := participant.SetMetadata(metadata); err != nil {
+			return err
+		}
 	}
 	if name != "" {
-		participant.SetName(name)
+		if err := participant.SetName(name); err != nil {
+			return err
+		}
```

Comment:
This does partial sets. Should we do that or do checks first and return on any limit excess and set all later so that it is all or nothing?

Reply:
yes, we changed oss to do this already

Reply:
The service path and client set path are split and not doing the same thing. I will make a change to unify it so that behaviour is consistent.